library/test/src/test/java/com/bumptech/glide/load/model/StreamEncoderTest.java

```diff
@@ -47,4 +51,46 @@ public void testWritesDataFromInputStreamToOutputStream() throws IOException {
 
     assertEquals(fakeData, new String(data, "UTF-8"));
   }
+
+  @Test
+  public void testFileNullException() throws IOException {
+    String fakeData = "SomeRandomFakeData";
+    final ByteArrayInputStream is = new ByteArrayInputStream(fakeData.getBytes("UTF-8"));
+    final boolean[] success = {false};
+
+      class EncodeNPETest implements ThrowingRunnable{
+        @Override
+        public void run() throws Throwable {
+          success[0] = encoder.encode(is, null, new Options());
+        }
+      }
+
+    // Write some data to #file making sure file.delete() executes properly
+    encoder.encode(is, file, new Options());
+
+
+    // Assert a NullPointerException is thrown here.
+    assertThrows(NullPointerException.class, new EncodeNPETest());
+
+    // Assert encoding process failed.
+    assertFalse(success[0]);
```

Comment:
This check is redundant, it's just asserting that the default value of success is false. I think you can remove this and just rely on the assertThrows above.